vertexai/genai/tokenizer/corpora_test.go

```diff
@@ -0,0 +1,305 @@
+		"Magahi-Agra":                    true,
+		"Bhojpuri-Agra":                  true,
+		"Esperanto-T61":                  true,
+		"Burmese_Myanmar-WinResearcher":  true,
+		"Armenian-DallakHelv":            true,
+		"Tigrinya_Tigrigna-VG2Main":      true,
+		"Amharic-Afenegus6..60375":       true,
+		"Navaho_Dine-Navajo-Navaho-font": true,
+		"Azeri_Azerbaijani_Cyrillic-Az.Times.Cyr.Normal0117": true,
+		"Azeri_Azerbaijani_Latin-Az.Times.Lat0117":           true,
+		"Czech-Latin2-err":     true,
+		"Russian_Russky-UTF8~": true,
+	}
+
+	return &udhrCorpus{
+		Encodings: encodings,
+		Skip:      skip,
+	}
+}
+
+// getEncoding returns the encoding for a given filename based on patterns
+func (ucr *udhrCorpus) getEncoding(filename string) (encoding.Encoding, bool) {
+	for _, pattern := range ucr.Encodings {
+		if pattern.Pattern.MatchString(filename) {
+			return pattern.Encoding, true
+		}
+	}
+	return nil, false
+}
+
+// shouldSkip checks if the file should be skipped
+func (ucr *udhrCorpus) shouldSkip(filename string) bool {
+	return ucr.Skip[filename]
+}
+
+// decodeBytes converts a byte slice from a specified encoding to a UTF-8 string
+func decodeBytes(enc encoding.Encoding, data []byte) (string, error) {
+	if enc == encoding.Nop {
+		return string(data), nil
+	}
+
+	if enc == nil {
+		return "", fmt.Errorf("unsupported encoding or custom handling required")
+	}
+
+	decoder := enc.NewDecoder()
+	reader := transform.NewReader(strings.NewReader(string(data)), decoder)
+	decodedBytes, err := io.ReadAll(reader)
+	if err != nil {
+		return "", fmt.Errorf("error decoding data: %v", err)
+	}
+
+	return string(decodedBytes), nil
+}
+
+const defaultModel = "gemini-1.0-pro"
+const defaultLocation = "us-central1"
+
+func TestCountTokensWithCorpora(t *testing.T) {
+	projectID := os.Getenv("VERTEX_PROJECT_ID")
+	if testing.Short() {
+		t.Skip("skipping live test in -short mode")
+	}
+
+	if projectID == "" {
+		t.Skip("set a VERTEX_PROJECT_ID env var to run live tests")
+	}
+	ctx := context.Background()
+	client, err := genai.NewClient(ctx, projectID, defaultLocation)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+	model := client.GenerativeModel(defaultModel)
+
+	t.Run("RemoteAndLocalCountTokensTest", func(t *testing.T) {
+		corporaUrl := "https://raw.githubusercontent.com/nltk/nltk_data/gh-pages/packages/corpora/udhr.zip"
+		fileCh, errCh := corporaGenerator(corporaUrl)
+		ucr := newUdhrCorpus()
+
+		// Iterate over files generated by the generator function
+		for fileInfo := range fileCh {
+			if ucr.shouldSkip(fileInfo.Name) {
+				fmt.Printf("Skipping file: %s\n", fileInfo.Name)
+				continue
+			}
+
+			enc, found := ucr.getEncoding(fileInfo.Name)
+			if !found {
+				fmt.Printf("No encoding found for file: %s\n", fileInfo.Name)
+				continue
+			}
+
+			decodedContent, err := decodeBytes(enc, fileInfo.Content)
+			if err != nil {
+				log.Fatalf("Failed to decode bytes: %v", err)
+			}
+
+			tok, err := New("gemini-1.5-flash")
```

Comment:
This model name is different from `defaultModel` -- shouldn't they be the same?

Reply:
Done